src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -2596,7 +2596,7 @@ out _copyLocalFiles
                 }
             }
 
-            MSBuildEventSource.Log.RarOverallStop();
+            MSBuildEventSource.Log.RarOverallStop(_resolvedFiles.Length, _resolvedDependencyFiles.Length, _copyLocalFiles.Length);
```

Comment:
Please consider to use null check/propagation, for example `_copyLocalFiles?.Length` at both Stop events callings. It is not guaranteed to be not null - especially in catch scenarios.

Reply:
Yes, great idea! That has bitten us before . . .